### ISSUE TITLE
Weird i18n language settings and fallbacks in Select2 4.0.3

### ISSUE THREAD

**Issue by cyrilmesvayn** [select2/select2#5046]
## Problems
- HTML lang attribute over-shadow global language configuration
- no default placeholder in i18n

## Expected
Select2 language should be loaded in "French"

## Results
Select2 language is "English"

## Steps
1. set `<html lang="en">`, or whatever else language on a page
2. insert jQuery 3.1.0
3. insert select2 4.0.3 css, then js, then i18n file, in that order.
4. insert script tag with either of the following: `$.fn.select2.defaults.set('language', 'fr');` OR using AMD :  `$.fn.select2.defaults.set('language', $.fn.select2.amd.require("select2/i18n/fr"));`
5. add select2 init : `$("#select2").select2({
    minimumInputLength: 2,
    placeholder: 'Test'
  });`

## Jsbin
https://jsbin.com/wofisagiqu/edit?html,output

## Notes
As per your [documentation](https://select2.org/i18n#language-files) :
this works as intended, however I find it strange that an explicit javascript declaration is overridden by a top-level lang attribute.

Also, a default placeholder value should be loadable from i18n file, and over-writable with the placeholder option at init time. I don't see in https://select2.org/placeholders something about this, so apologies if it's already possible and I don't know about it.

Thanks for your consideration.

**Comment by alexweissman** [comment]
TL;DR: A declaration like `$.fn.select2.defaults.set('language', 'fr');` should override `<html lang="en">`, but it doesn't.

**Comment by Ketcap** [comment]
+1 i got the same error on meteor and it crushes my app so changed version '4.0.3' from '^4.0.3'

**Comment by alexweissman** [comment]
It looks like #4887 might fix this?

**Comment by alexweissman** [comment]
@Ketcap wait, does that mean that this has been fixed in a more recent version?

**Comment by tuanpht** [comment]
Still same problem in version 4.0.5, I have to remove html `lang` attribute` :(

**Comment by tuanpht** [comment]
Finnally, I can customize message using this:
```js
    const select2LangJa = $.fn.select2.amd.require('select2/i18n/ja');

    select2LangJa.errorLoading = function () {
        return '';
    }
```

**Comment by kevin-brown** [comment]
Is this still a problem in the latest 4.0.6 release?

This would benefit from a failing test case. The fallback to the `lang` attribute is well documented, but we may want to adjust the order so that it treats any defaults first before falling back to it.

**Comment by kevin-brown** [comment]
I believe this is a duplicate of https://github.com/select2/select2/pull/5602 and was resolved in 4.0.9.
